### PR TITLE
New methods for Bot 2.1 API

### PR DESCRIPTION
**Pull request by leandrotoledo**
List of new methods and details about 2.1 API here #302 

Main bullet points as follows:
- pre-commit doesn't check yapf style in examples folder
- added `**kwargs` to all API calls
- new error `BadRequest` (inherits `NetworkError` for backwards compatibility)
- changing order in factory `InputMessageContent` (because `**kwargs` may create a `InputLocationContent` instead of `InputVenueContent`)
- tests (by @rahiel - kudos!)
- test if fail for `testLeaveChat` (due the nature of the test)

Assigning to @jh0ker

### CODE REVIEW

File: telegram/bot.py [hunk 1]
```diff
@@ -1302,3 +1462,8 @@ def __reduce__(self):
     edit_message_reply_markup = editMessageReplyMarkup
     get_updates = getUpdates
     set_webhook = setWebhook
+    leave_chat = leaveChat
+    get_chat = getChat
+    get_chat_administrators = getChatAdministrators
+    get_chat_member = getChatMember
+    get_chat_member_count = getChatMembersCount
```
Comment (jh0ker) [inline telegram/bot.py]: should be `get_chat_members_count`

**Comment by jh0ker** [comment]
LGTM!

**Comment by jh0ker** [comment]
However, we need a new Handler class for Updates with `edited_message`

**Comment by leandrotoledo** [comment]
@jh0ker can't we?

```
    @staticmethod
    def text(update):
+       message = update.message or update.edited_message
        return message.text and not message.text.startswith('/')

    @staticmethod
    def command(update):
+       message = update.message or update.edited_message
        return message.text and message.text.startswith('/')

...

        if isinstance(update, Update) and (update.message +or update.edited_message):
```

Only text seems to be editable.

**Comment by leandrotoledo** [comment]
Uhm, or maybe edited_text or edited_command, thus it won't break any Handlers parsing update.message currently.

**Comment by jh0ker** [comment]
Yes that's good! But then we still have that problem for the command handler...

**Comment by leandrotoledo** [comment]
a0bb573 LGTM.

I think this is good to be merged.

**Comment by leandrotoledo** [comment]
Tests were failing because test_filters still had Update as argument for filters, should be fixed on 80fbe98

**Pull request merged**